src/witness_gen.rs

```diff
@@ -146,16 +170,74 @@ fn map_to_witness_text<'query>(
     }
 }
 
+/// Generates a single witness crate
+fn generate_witness_crate(
+    witness_set_dir: &Path,
+    witness_name: &str,
+    index: usize,
+    _witness_text: String,
```

Comment:
Possibly `&str` here instead? I don't think we care about having ownership particularly strongly.